api/v1alpha1/instaslice_types.go

```diff
@@ -56,105 +33,166 @@ const (
-	Start int32 `json:"start"`
-	// size of profile that begins from start position
-	// +required
-	Size int32 `json:"size"`
-	// podUUID represents uuid of user workload
-	// +required
-	PodUUID string `json:"podUUID"`
-	// gpuUUID represents gpu uuid of selected gpu
-	// +required
+
+	// resources specifies resource requirements for the allocation
+	// +kubebuilder:validation:optional
+	Resources corev1.ResourceRequirements `json:"resources"`
+
+	// podRef is a reference to the gated Pod requesting the allocation
+	// +kubebuilder:validation:optional
+	PodRef corev1.ObjectReference `json:"podRef"`
+}
+
+type AllocationResult struct {
+	// migPlacement specifies the MIG placement details
+	// +kubebuilder:validation:Required
+	MigPlacement Placement `json:"migPlacement"`
+
+	// gpuUUID represents the UUID of the selected GPU
+	// +kubebuilder:validation:Required
 	GPUUUID string `json:"gpuUUID"`
-	// nodename represents name of the selected node
-	// +required
-	Nodename string `json:"nodename"`
-	// allocationStatus represents status of allocation
-	// +kubebuilder:validation:Enum:=deleted;deleting;ungated;creating;created
-	// +required
-	Allocationstatus AllocationStatus `json:"allocationStatus"`
-	// resourceIdentifier represents uuid used for creating configmap resource
-	// +required
-	Resourceidentifier string `json:"resourceIdentifier"`
-	// namespace represents namespace of user workload
-	// +required
-	Namespace string `json:"namespace"`
-	// podName represents name of the user workload pod
-	// +required
-	PodName string `json:"podName"`
-	// cpu represents amount of cpu requested by user workload
-	// +required
-	Cpu int64 `json:"cpu"`
-	// memory represents amount of memory requested by user workload
-	// +required
-	Memory int64 `json:"memory"`
+
+	// nodename represents the name of the selected node
+	// +kubebuilder:validation:Required
+	Nodename types.NodeName `json:"nodename"`
+
+	// allocationStatus represents the current status of the allocation
+	// +kubebuilder:validation:Enum=deleted;deleting;ungated;creating;created
+	// +kubebuilder:validation:Required
+	AllocationStatus AllocationStatus `json:"allocationStatus"`
+
+	// configMapResourceIdentifier represents the UUID used for creating the ConfigMap resource
+	// +kubebuilder:validation:Required
+	ConfigMapResourceIdentifier types.UID `json:"configMapResourceIdentifier"`
+
+	// conditions provide additional information about the allocation
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+type DiscoveredGPU struct {
+	// gpuUUID represents the UUID of the GPU
+	// +kubebuilder:validation:Required
+	GPUUUID string `json:"gpuUuid"`
+
+	// gpuName represents the name of the GPU
+	// +kubebuilder:validation:Required
+	GPUName string `json:"gpuName"`
+
+	// gpuMemory represents the memory capacity of the GPU
+	// +kubebuilder:validation:Required
+	GPUMemory resource.Quantity `json:"gpuMemory"`
+}
+
+type DiscoveredNodeResources struct {
```

Comment:
@harche , According to this [comment](https://github.com/openshift/instaslice-operator/pull/421#discussion_r1941678587), we should also add a field in the `InstasliceStatus` to persist the Node's boot id in order to identify a node reboot.

cc: @rphillips

Reply:
It's one such solution. @sairameshv Do you have a design on detecting the reboot for the daemonset?

Reply:
I was mainly relying on the Node NotReady condition from the Node object's status. Other than that, I don't have any idea yet